include/node.hpp

```diff
@@ -613,17 +614,19 @@ class node : protected std::tuple<Arguments...> {
             }
 
             if (numerator != 1. || denominator != 1.) {
```

Comment:
Following the C++ principle "only pay for what you use": could you add a constexpr-if guard that the stride and ratio checks are only performed when the node carries the corresponding NTTP tag? The rationale is that we want to pay for the checks only when the user actually requires the decimation/interpolation/stride, which is probably only for <5% of all blocks ...

Also, could you create an issue describing this and other settings pre-condition check functionalities that keep a sub-set of block settings consistent and/or potentially throws a warning to the user and/or rejects accepting the new incompatible settings? We will need to address/discuss this with a wider audience. To note: @mormj

Reply:
Done.
And here is the issue: https://github.com/fair-acc/graph-prototype/issues/146